### PR TITLE
Bug 1904380 - Forwarding logs to Kafka using Chained certificates fails with error "state=error: certificate verify failed (unable to get local issuer certificate)

### PR DESCRIPTION
**Pull request by syedriko**
### Description

Bug 1904380 - Forwarding logs to Kafka using Chained certificates fails with error "state=error: certificate verify failed (unable to get local issuer certificate)
- Upgraded Kafka to 2.7.0
- Switched the kafka logforwarding test to run over TLS
- The Kafka broker certificate chain is Server - Intermediate CA - Root CA
- Added Java Key Store generation functions

/cc @alanconway 
/assign @jcantrill 

### Links
- Depending on PR(s): https://github.com/openshift/origin-aggregated-logging/pull/2068
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1904380

### CODE REVIEW

**Comment by syedriko** [comment]
@jcantrill Could you bless this

File: test/helpers/kafka/broker.go [hunk 1]
```diff
@@ -138,7 +141,7 @@ func NewBrokerStatefuleSet(namespace string) *apps.StatefulSet {
 					Containers: []v1.Container{
 						{
 							Name:  kafkaBrokerContainerName,
-							Image: "solsson/kafka:2.4.1",
+							Image: "solsson/kafka@sha256:754cd2e82bbddd37a81d214be33fd15eb429913f9ac51dedf89d446876b95ffc",
```
[the remark this answers is not in the document]
Comment (syedriko) [reply test/helpers/kafka/broker.go:144]: There's no tag for 2.7.0, it's just "latest". Between latest and sha I picked sha.

[the remark this answers is not in the document]
Comment (syedriko) [reply test/helpers/kafka/broker.go:144]: I'm not actually sure it's a good idea to rely on images from Docker Hub now that they introduced rate limiting. We could start with copying these to quay. Or better yet switch to Strimzi.

[the remark this answers is not in the document]
Comment (jcantrill) [reply test/helpers/kafka/broker.go:144]: if you have concerns, feel free to push to `quay.io/openshift-logging`.  Seems reasonable for us to have a controlled copy of the image against which we test.

File: test/helpers/kafka/zookeeper.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func NewZookeeperStatefuleSet(namespace string) *apps.StatefulSet {
 					InitContainers: []v1.Container{
 						{
 							Name:  "init-config",
-							Image: "solsson/kafka-initutils@sha256:f6d9850c6c3ad5ecc35e717308fddb47daffbde18eb93e98e031128fe8b899ef",
+							Image: "quay.io/openshift-logging/kafka-initutils:2.7.0",
```
Comment (jcantrill) [inline test/helpers/kafka/zookeeper.go:80]: Extract to constant?"

File: test/helpers/kafka/zookeeper.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func NewZookeeperStatefuleSet(namespace string) *apps.StatefulSet {
 					Containers: []v1.Container{
 						{
 							Name:  zookeeperDeploymentName,
-							Image: "solsson/kafka:2.4.1",
```
Comment (jcantrill) [inline test/helpers/kafka/zookeeper.go:104]: Constant?

**Review by jcantrill — CHANGES_REQUESTED** [review]

**Comment by syedriko** [comment]
@jcantrill yea? nay?

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**